hazelcast/src/main/java/com/hazelcast/internal/partition/operation/BaseMigrationOperation.java

```diff
@@ -107,25 +114,66 @@ private void verifyPartitionStateVersion() {
         }
     }
 
+    /** Verifies that the local master is equal to the migration master. */
+    final void verifyMaster() {
+        NodeEngine nodeEngine = getNodeEngine();
+        Address masterAddress = nodeEngine.getMasterAddress();
+
+        if (!migrationInfo.getMaster().equals(masterAddress)) {
+            throw new IllegalStateException("Migration initiator is not master node! => " + toString());
+        }
+
+        if (getMigrationParticipantType() == MigrationParticipant.SOURCE && !masterAddress.equals(getCallerAddress())) {
+            throw new IllegalStateException("Caller is not master node! => " + toString());
+        }
+    }
+
     /**
-     * Checks if the local UUID matches the migration source or destination UUID if this node is the migration source or
+     * Checks if the local member matches the migration source or destination if this node is the migration source or
      * destination.
      */
-    private void verifyMemberUuid() {
+    private void verifyMigrationParticipant() {
         Member localMember = getNodeEngine().getLocalMember();
-        if (localMember.getAddress().equals(migrationInfo.getSource())) {
-            if (!localMember.getUuid().equals(migrationInfo.getSourceUuid())) {
+        if (getMigrationParticipantType() == MigrationParticipant.SOURCE) {
+            if (migrationInfo.getSourceCurrentReplicaIndex() == 0
+                    && !migrationInfo.getSource().isIdentical(localMember)) {
                 throw new IllegalStateException(localMember
-                        + " is the migration source but has a different UUID! Migration: " + migrationInfo);
+                        + " is the migration source but has a different identity! Migration: " + migrationInfo);
             }
-        } else if (localMember.getAddress().equals(migrationInfo.getDestination())) {
-            if (!localMember.getUuid().equals(migrationInfo.getDestinationUuid())) {
+            verifyPartitionOwner();
```

Comment:
Yes, moved from `MigrationRequestOperation`. Source doesn't need to know the destination... But this might be a stale migration request and the destination might be off already. This check is to avoid creation of expensive replication operations when destination is not known. Since invocation will fail if that's the case.